### PR TITLE
Remove ROS arguments before passing to QApplication

### LINKED ISSUE #442: rviz2 crashes when launched as a node
When launching using a `launch_ros.actions.Node`, RViz2 crashes, choking on the newly introduced `--ros-args` option. Seems to be caused by https://github.com/ros2/launch_ros/pull/52

```
from pathlib import Path

from ament_index_python.packages import get_package_share_directory
from launch import LaunchDescription
from launch.actions import SetEnvironmentVariable, DeclareLaunchArgument
from launch.substitutions import LaunchConfiguration
from launch_ros.actions import Node


def generate_launch_description():
    rviz_config = Path(get_package_share_directory('openrover_demo'), 'config', 'default.rviz').resolve()
    assert rviz_config.is_file()

    return LaunchDescription([
        DeclareLaunchArgument('frame', default_value='map', description='The fixed frame to be used in RViz'),
        SetEnvironmentVariable('RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED', '1'),
        Node(
            package='rviz2', node_executable='rviz2', output='screen',
            arguments=[
                '--display-config', str(rviz_config),
                '--fixed-frame', LaunchConfiguration(variable_name='frame')
            ]
        )
    ])
```

```
$ ros2 launch openrover_demo rviz.launch.py
[INFO] [launch]: All log files can be found below /home/dan/.ros/log/2019-08-07-14-17-41-837727-rhea-5588
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [rviz2-1]: process started with pid [6352]
[rviz2-1] Unknown option 'ros-args'.
[ERROR] [rviz2-1]: process has died [pid 6352, exit code 1, cmd '/home/dan/ros2_ws/install/lib/rviz2/rviz2 --display-config /home/dan/ros2_ws/install/share/openrover_demo/config/default.rviz --fixed-frame map --ros-args'].
```

### PR DESCRIPTION
**Pull request by jacobperron**
Resolves #442

### CODE REVIEW

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8516/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4399/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6933/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8430/)

**Comment by dirk-thomas** [comment]
In ROS 1 the [init(int argc, char ** argv)](http://docs.ros.org/api/roscpp/html/namespaceros.html#a7f5c939b8a0548ca9057392cc78d7ecb) function not only used the ROS specific arguments but also updated the arguments to leave only the non-ROS arguments. I think a similar function would be very helpful to avoid the overhead from this PR.

File: rviz2/src/main.cpp [hunk 1]
```diff
@@ -40,7 +41,15 @@
 
 int main(int argc, char ** argv)
 {
-  QApplication qapp(argc, argv);
+  // remove ROS arguments before passing to QApplication
+  std::vector<std::string> non_ros_args = rclcpp::remove_ros_arguments(argc, argv);
+  std::vector<char *> non_ros_args_c_strings;
```
Comment (wjwwood) [inline rviz2/src/main.cpp:46]: Do these `char *`'s need to be non-`const`?

**Review by wjwwood — APPROVED** [review]
lgtm, thanks for tackling it Jacob.

File: rviz2/src/main.cpp (hunk 1 above)
Comment (jacobperron) [reply rviz2/src/main.cpp:46]: Yes, I think so. I believe the compiler was complaining about discarding const-ness when passing to QApplication.

**Comment by jacobperron** [comment]
> In ROS 1 the init(int argc, char ** argv) function not only used the ROS specific arguments but also updated the arguments to leave only the non-ROS arguments. I think a similar function would be very helpful to avoid the overhead from this PR.

Sounds reasonable to me. But we should be careful in this case, since we still want to pass the argv containing ROS arguments later in `main`:

https://github.com/ros2/rviz/blob/265dbb345584aff51c2586daba0c8f6e2aff4c21/rviz2/src/main.cpp#L75

**Comment by wjwwood** [comment]
Also see: https://github.com/ros2/rcl/issues/219

**Pull request merged**